Khan/react-redux
# FEI-4163: Update react-redux to check wonder-blocks' Server.isServerSide()

## Summary:
We're in the process of updating webapp to use react-redux 7.2.6.  Unfortunatley, the logic that this version uses to determine whether we're SSR-ing or not doesn't work with how we SSR things.  As a result, it ends up using 'useLayoutEffect' instead of 'useEffect' when SSR-ing and that results in hydration errors.

This PR fixes the issue by calling 'Server.isServerSide()' from wonder-blocks-core which is how we determine if we're SSR-ing or not.

It would be nice if react-redux provide a hook to customize this check.  In the meantime though it's easier for us to just modify the check ourselves.

Issue: FEI-4163

## Test plan:
- yarn build
- yarn test

```
kevinbarabash@Kevins-MacBook-Pro react-redux % yarn test
yarn run v1.22.11
$ npm run lint
npm WARN lifecycle The node binary used for scripts is /var/folders/nc/zq7cwh1s5wd6njvp_89zt3d00000gn/T/yarn--1638483384584-0.40320656906122343/node but npm is using /Users/kevinbarabash/.nvm/versions/node/v12.22.5/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> react-redux@7.2.6 lint /Users/kevinbarabash/khan/react-redux
> eslint src test/utils test/components test/hooks

$ jest
 PASS   ReactDOM  test/utils/Subscription.spec.js
 PASS   ReactDOM  test/utils/shallowEqual.spec.js
 PASS   ReactDOM  test/utils/isPlainObject.spec.js
 PASS   ReactDOM  test/hooks/useReduxContext.spec.js
 PASS   ReactDOM  test/integration/server-rendering.spec.js
 PASS   ReactDOM  test/hooks/useDispatch.spec.js
 PASS   ReactDOM  test/integration/dynamic-reducers.spec.js
 PASS   ReactDOM  test/components/connectAdvanced.spec.js
 PASS   ReactDOM  test/hooks/useSelector.spec.js
 PASS   ReactDOM  test/components/hooks.spec.js
 PASS   ReactDOM  test/components/Provider.spec.js
 PASS   ReactDOM  test/components/connect.spec.js (6.057 s)
[BABEL] Note: The code generator has deoptimised the styling of undefined as it exceeds the max of 500KB.
 PASS   React Native  test/react-native/batch-integration.js (11.603 s)

Test Suites: 13 passed, 13 total
Tests:       5 skipped, 134 passed, 139 total
Snapshots:   0 total
Time:        14.288 s
Ran all test suites in 2 projects.
✨  Done in 20.85s.
```

- yarn pack
- copy the tarball to webapp/services/sattic/node_modules
- unpack the tarball and rename 'package' to 'react-redux'
- run the route tests, see that they pass
